Executor CLI Volume Enhancements

akutz opened the pull request:
This path adds the following commands to the executor:

* volumeCreate
* volumeRemove
* volumeAttach
* volumeDetach

Now storage platforms can participate in the above workflow logic on the client-side without being part of the libStorage client.

codenrhoden commented on api/types/types_drivers_executor.go:
```diff
@@ -46,6 +46,18 @@ const (
+	LSXCmdVolumeAttach = "volumeAttach"
+
+	// LSXCmdVolumeDeatch is the command for detaching a volume.
+	LSXCmdVolumeDeatch = "volumeDetach"
```
s/VolumeDeatch/VolumeDetach/

akutz replied:
Hi @codenrhoden,

Sorry, this is labelled a work-in-progress, but I only updated the situation in Slack where I mentioned the PR is incomplete.

codenrhoden replied:
Ah yes, it clearly does say that. Apologies. That would be the answer to "what am I missing"?  😄

akutz replied:
![image](https://cloud.githubusercontent.com/assets/101085/23135842/bdb656a0-f75f-11e6-9bfa-96b71644e034.png)

akutz replied:
Hi @codenrhoden,

Sorry, I wasn't trying to be snarky. I had not refreshed the page and did not see your follow-up. The kids do not have school, and I'm busy watching She-Ra with my daughter while my son is at speech therapy. :)

codenrhoden reviewed (changes requested):
Looks good, but seeing this type makes me realize that none of the new consts created, `LSXCmdVolume{Create,Remove,Attach,Detach}` are actually used anywhere. What am I missing?

akutz commented:
Hey @codenrhoden,

Now the executor should be complete.

akutz commented:
As this PR isn't required for this release, I will be changing this PR's base to `master` once libStorage 0.5.0 is GA.

akutz committed:
Executor CLI Volume Enhancements

This path adds the following commands to the executor:

* volumeCreate
* volumeRemove
* volumeAttach
* volumeDetach

Now storage platforms can participate in the above workflow logic on the
client-side without being part of the libStorage client.
